src/components/TreeGrid.ts

```diff
@@ -46,13 +46,65 @@ const wrapContent =
     state: 'closed',
   });
 
+const hierarchyMapToRows = <T extends RowContents>(hierarchyMap: Map<number, T[]>, domainOptionRoot: T): Row<T>[] => {
+  // initialize an array of rows
+  const rows: Row<T>[] = [];
+  const depth = 0;
+
+  const traverseHierarchy = (parent: T, depth: number) => {
+    // push the parent row
+    rows.push({
+      contents: parent,
+      depth,
+      isFetched: true,
+      state: 'open',
+    });
+    // if the parent has Children
+    if (parent.hasChildren) {
+      // get the children
+      const children: T[] | undefined = hierarchyMap.get(parent.id);
+      // if children is undefined, return nothing
```

Comment:
Maybe add a comment about how this is expected; we only have children for expanded nodes, so we can expect to get nodes that have children but don't have the children in `hierarchyMap`.